### DISCUSSION TITLE
Database cleanup

### DISCUSSION THREAD

**Discussion by andyrozman** [NightscoutFoundation/xDrip#3605] [Q&A] [answered]
Is there a way to cleanup the Xdrip database... I have been using Xdrip for 10 years now, so my database has grown (google backup was about 150 Mb, I almost thought it was broken, backup upload took so long). I would like to remove all data that is more than 2 years old (or something like that).

**Comment by Navid200** [comment, marked as answer]
Go to Settings > Less common settings > Glucose retention  

If you have it set to 0, it never deletes anything.
Set it to 730 if you want it to keep 2 years worth of data.

**Comment by andyrozman** [comment]
Mhm. So I set this, but when I had next cloud backup, the file uploaded was more than previous one, so database was not reduced. Do I need to run any action after I have set this values?

**Reply by Navid200** [reply]
I believe this has to do with the daily service.  So, if I'm not mistaken, the old data is deleted only once a day.  
Has there been a day, or more, since you set it?

**Reply by andyrozman** [reply]
Yes it was... I just waited... 5 days have passed, and data is still there... When I do Cloud backup backup file is still 155 Mb... Automatic Cloud backups might be broken because of that (I have set daily backup) and last backup was when I did it manually (and it takes forever with such big file).

**Reply by Navid200** [reply]
I will do a test and report.

**Reply by Navid200** [reply]
The backup mechanism allows you to restore to a time in the past.  So, when you perform a backup, it does not overwrite the backup.  It adds the backup to Google Drive as a revision.
So, if you look at the size of the backup file on Google Drive, it is not supposed to become small.  it is actually supposed to gradually grow.
If I am not mistaken, it will delete a 100-day old revision.

But, the backup process should not be slow.

May be you can clarify what you mean by "data is still there".

**Reply by Navid200** [reply]
Can I ask you to do something?  

Use a computer and log into the Google account you use for Google Drive backup.
Go to Google Drive.
Expand to see the files on the drive.
There should be a folder called xDrip-Backups.  Expand the folder.
You should see the names of your phones you have used Google Drive backup with.  If you have used it only with one phone, you will only see that phone.
You should see something like this:  
![Screenshot 2024-08-09 115328](https://github.com/user-attachments/assets/0d1443f4-93c7-4769-8a79-6cf5cbd44d21)
 
Click on the 3 dots menu on the right: 
![Untitled](https://github.com/user-attachments/assets/cd5fcee6-0afa-415d-a903-28543d3563b2)

You should see something like this for versions.  
![Screenshot 2024-08-09 115746](https://github.com/user-attachments/assets/5ea3b913-39fc-42da-9a42-a5bc72a8621f)
  
We have not added the ability to choose which one to restore yet.
At this point, if you absolutely have to do that, you will need to manipulate the backup and change the active version.  

Can you look at this and see if it explains why the size has not gone down yet.

**Reply by Navid200** [reply]
Alternatively, you can use a voice command to see what your database size is.
https://navid200.github.io/xDrip/docs/Settings/VoiceCommands.html  

The voice command is database administration.  
After you enter it, you will see a menu with 4 options.  You should use neither!  The size is shown under the menu.

For the record voice commands are meant for experienced users or developers only.  
You could do damage if you are not careful.  That's why it is a voice command.
